guides/source/action_controller_overview.md

```diff
@@ -666,6 +668,128 @@ end
+```
+
+Note that this would only affect the +#index+ action.
+
+Two, by knowing that when the +JSON+ renderer is added (<tt>ActionController::Renderers.add :json, &block</tt>),
+that the +_render_with_renderer_json+ method is generated, and knowing that the default Renderer does not modify strings,
+we can override that method as follows:
+
+```diff
+class UsersController < ApplicationController
+
++   def _render_with_renderer_json(json, options)
++     if json.is_a?(String)
++       json
++     else
++       json = json.as_json(options) if json.respond_to?(:as_json)
++       JSON.pretty_generate(json, options)
++     end
++     super(json, options)
++   end
+end
+```
+
+This customization will affect the <tt>UsersController</tt> and its children.
+
+Note that we needed to call <tt>super(json, options)</tt> in order to return control back
+to the Renderer.
+
+Adding a new Renderer
+---------------------------
+
+Let's say we wanted to add a +CSV+ rendering to our application.
+
+In order to be able to <tt>render csv: @csv</tt>, we need to add a +CSV+ Renderer.
+
+Adding a +CSV+ renderer is as simple as <tt>ActionController::Renderers.add :csv, &block</tt>,
+where the block sets any response attributes and returns the response body.
+
+An simple implementation may look like:
+
+```ruby
+ActionController::Renderers.add :csv do |csv, options|
+  filename = options[:filename] || 'data'
+  str = csv.respond_to?(:to_csv) ? csv.to_csv : csv.to_s
+  send_data str, type: Mime[:csv],
+    disposition: "attachment; filename=#{filename}.csv"
+end
+```
+
+This will generate the +_render_with_renderer_csv+ in controllers, to be used when the key +:csv+
+is passed as an option to +render+.
+
+Note, also, that we're using <tt>ActionController::DataStreaming#send_data</tt> to force a download.
+
+If we wanted to make it possible to configure how we generate the +CSV+ string, using the +serializing+
+macor, we can write the renderer as:
+
+```diff
++ ActionController::Renderers.add_formatter :csvdo |json, options|
++   csv.respond_to?(:to_csv) ? csv.to_csv : csv.to_s
++ end
+ActionController::Renderers.add :csv do |csv, options|
+  filename = options[:filename] || 'data'
+-   str = csv.respond_to?(:to_csv) ? csv.to_csv : csv.to_s
++   str = _formatters[:csv].call(csv, options)
+  send_data str, type: Mime[:csv],
+    disposition: "attachment; filename=#{filename}.csv"
+end
+```
+
+In this case, we did not need to register <tt>Mime[:csv]</tt>, since it is already registered
+by Rails.
+
+Rails uses the registered mime type to:
+
+1. Set the +response.content_type+ (<tt>Content-Type</tt>).
+1. Recognize the request +format+ (<tt>Content-Type</tt>), and parse accordingly.
+1. Negotiate what +request.formats+ (<tt>Accept</tt>) to +respond_to+.
+1. Map a url ending with a mime type's name to its template or renderer.
+
+See <tt>ActionController::Renderers.use_renderers</tt> for more information.
+
+Should any of this go in guides/source/layouts_and_rendering.md ?
```

Comment:
In fact I'd say this whole section should be there. This guide should only be a overview and we are going to deep on it.